src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1525,8 +1525,15 @@ def update_stack_component(
                         "Service connector with ID "
                         f"{component_update.connector} not found."
                     )
+            else:
+                existing_component.connector = None
+                existing_component.connector_resource_id = None
```

Comment:
Currently, if any value in an update model is `None`, it means that this value was not provided and should not be updated. So I think the current design is breaking this convention since `connector=None` now disconnects the existing connector and thereby leads to updates.

Instead, I would suggest to add `disconnect` as a field of `StackComponentUpdateModel` explicitly and then use this here instead:

```suggestion
            elif component_update.disconnect:
                existing_component.connector = None
                existing_component.connector_resource_id = None
```

Reply:
I think it's fine as it is, because this immediately fixes the UI as well. The alternative also requires dashboard changes. If it helps, this is the same approach used for service connector updates: some update fields are "special", in the sense that if omitted from the update request, they are interpreted as a value removal rather than a value omission.